src/components/tools/tile-comments.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from "react";
+import { observer, inject } from "mobx-react";
+import { BaseComponent } from "../base";
+import { TileCommentsModelType } from "../../models/tools/tile-comments";
+import { IToolApiMap } from "./tool-tile";
+
+import "./tile-comments.sass";
+
+interface IProps {
+  model: TileCommentsModelType;
+  toolApiMap?: IToolApiMap;
+}
+
+@inject("stores")
+@observer
+export class TileCommentsComponent extends BaseComponent<IProps, {}> {
+
+  public render() {
+    const { model } = this.props;
+    if (!model.visible && model.comments.length) {
+      return this.renderClosed();
+    }
+    const { class: clazz } = this.stores;
+    return (
+      <div className="tile-comments open">
+        <div className="close" title="Close" onClick={this.closeComments}>
+          <svg className={`icon icon-delete-tool`}>
+            <use xlinkHref={`#icon-delete-tool`} />
+          </svg>
+        </div>
+        <div className="comments">
+          <div className="title">Comments</div>
+          {
+            model.comments.map(comment => {
+              const student = clazz.getUserById(comment.uid);
+              const name = student ? student.displayName : "Student";
+              return (
+                <div className="comment" key={comment.key}
+                     onMouseEnter={this.handleHover(comment.selectionInfo)}
+                     onMouseLeave={this.handleLeave(comment.selectionInfo)}>
+                  {`${name}: ${comment.text}`}
+                </div>
+              );
+            })
+          }
+        </div>
+      </div>
+    );
+  }
+
+  private handleHover = (selectionInfo?: string) => () => {
+    const { toolApiMap } = this.props;
+    const toolApi = toolApiMap && toolApiMap[this.props.model.tileId];
+    if (toolApi && selectionInfo) {
```

Comment:
I'm passing the `toolApiMap` all the way down to the tiles to do this...something feels wrong about passing both the `toolApiInterface` and `toolApiMap` everywhere, would you have approached this differently @kswenson?

Reply:
I agree it feels like exposing an implementation detail. In this particular instance, I think it would work to pass down a callback like:
```
onHighlightSelection(tileId, selectionInfo, isSelected)
```
In other contexts in which more of the API is required, it may not be as easy to work around. Another approach would be to pass down the function of the map as a callback, e.g.
```
onGetToolApi(tileId: string)
```
which at least encapsulates the form of the implementation detail if not its existence. In the end a combination of approaches might be required.